lib/engines/content_block_manager/test/components/content_block/edition/details/fields/array/array_item_component_test.rb

```diff
@@ -32,30 +34,52 @@ class ContentBlockManager::ContentBlockEdition::Details::Fields::Array::ItemComp
     let(:field_value) { ["", "Some text"] }
     let(:index) { 1 }
 
-    it "renders a text field" do
-      render_inline(component)
+    describe "if the item can be deleted" do
+      let(:can_be_deleted) { true }
 
-      assert_selector "label", text: "Bar"
-      assert_selector "input[type='text'][value='Some text'][name='foo[bar][]'][id='foo_bar_1']"
-    end
+      it "renders with the correct class" do
```

Comment:
perhaps it's worth being explicit here about the presence / absence of the `--immutable` class and it's purpose? e.g.

```
it "renders with the '--immutable' class to display the 'delete' button" do
```